src/ORM/Hierarchy/Hierarchy.php

```diff
@@ -23,21 +23,6 @@
  */
 class Hierarchy extends DataExtension
 {
-
-    /**
-     * Cache for {@see numChildren()}
-     *
-     * @var int
-     */
-    protected $_cache_numChildren = null;
-
-    /**
-     * Cache for {@see Children()}
-     *
-     * @var SS_List
-     */
-    protected $_cache_children = null;
```

Comment:
I’ve done some provisional benchmarking of a blank install, replacing this with a static cache in `Hierarchy` (e.g. `if (isset(static::$_cache_children[$this->owner->ID]))`) and it seems to save a few calls to `->stageChildren()` when looking the `treeview` AJAX request. I can’t figure out why (perhaps individual pages are being garbage collected and then re-fetched?), do you want to look at that?

Usual caveat about memory usage may or may not apply... I’m struggling to think of the memory usage implications of prototype extension vs static cache vs cache on owner

Reply:
One problem is that it may be more than a simple ID to key the cache; It could also be stage.

Reply:
Good point